tests: kernel: Convert test case to use ztest

punitvara opened the pull request:
This patch reduces unnecessary output on console. Those things
are replaced by ztest framework APIs

Signed-off-by: Punit Vara <punit.vara@intel.com>

nashif commented on tests/kernel/errno/src/0001-tests-kernel-Convert-test-case-to-use-ztest.patch:
```diff
@@ -0,0 +1,444 @@
+From c8556181b709816a7b47957772a9eaea65c938a3 Mon Sep 17 00:00:00 2001
```
remove this file

punitvara replied:
Done

nashif reviewed (changes requested):
please split the commit into multiple commits, one for each test you are changing.

punitvara committed:
tests: critical: Convert testcase into ztest

This commit uses ztest framework APIS to make ouput unified
with other test cases.

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: errno: convert test case to use ztest

This patch convert normal test case to use ztest framework
APIs and remove unnecessary output.

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: irq_offload: Convert test case to use ztest

This patch reduces unnecessary output on console. Those things
are replaced by ztest framework APIs

Signed-off-by: Punit Vara <punit.vara@intel.com>

nashif commented:
Please add the test name into the title of the commit message and make them unique, i.e

tests: errno: ....
tests: critical: ...

Vudentz commented on tests/kernel/critical/src/Makefile:
```diff
@@ -1,4 +1,3 @@
 include $(ZEPHYR_BASE)/tests/Makefile.test
-
-obj-y = critical.o
+obj-y = main.o test_critical_context.o
```
I don't really get it why we need to split the test definition and tests into different files.

punitvara replied:
@Vudentz  Some test cases have their own name in src/ directory like critical test case has file name critical.c , mslab test case has file name slab.c. All these looks different and messy. If we can have one main file and other file (whatever name is) , all the test case would looks same and easy to understand structure for any one.

nashif replied:
remove test_ prefix from files

punitvara replied:
Done

nashif replied:
We do not need to split every testcase, splitting only has an advantage when you are testing different topics in one testcase and need the logical separation to make it easier to read and understand. Simple tests can be kept in one file.

nashif commented on tests/kernel/errno/prj.conf:
```diff
@@ -1 +1,2 @@
 # nothing yet
```
remove that comment

punitvara replied:
Done

nashif commented on tests/kernel/errno/src/test_errno_context.c:
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2015 Wind River Systems, Inc.
```
add new line for copyright with Intel.

punitvara replied:
Done
